book-2-mvc/chapters/INTRO_TO_MVC.md

```diff
@@ -0,0 +1,126 @@
+
+### Client/Server (React and JSON Web API)
+1. A user wants to see all the animals in Kennel, so they open their browser and make a request to `https://kennel.nss.com` in the url bar
+
+1. The request gets handled by an nss server, and the server responds by sending all the react code back. Because it's a react application, almost all the code is javascript.
+
+1. The browser immediately starts executing the react js code. Part of that code tells the browser to make `fetch` a fetch call to `https://kennel.nss.com/api/animals` so the react app can get animal data in the form of JSON. If we were to pause this whole process right here, the user would see _some_ of the animal page, but would not see any animals listed since that data has not been returned yet.
+
+1. The nss server receives the request made by the react app for animal data. It makes a call to the database to ask for all the saved information on the animals. The nss server then sends a response to the react app with all the animal information as JSON.
+
+1. The react application now has all the animal information and can finish filling out the page by mapping each animal object to a visual component.
+
+
+### MVC Pattern
+
+1. A user wants to see all the animals in Kennel, so they open their browser and make a request to `https://kennel.nss.com` in the url bar
+
+1. The request gets handled by an nss server. The code running on the server knows that the user wants to see the web page with all the animals on it, so the server makes a call to the database to ask for all the saved information on the animals.
+
+1. The server then takes the animal data returned by the database, and dynamically creates an html page.
+
+1. The server sends the generated html page back to the browser along with whatever css and js is needed.
+
+### Which is better?
+The big difference here is that in the Client/Server pattern, most of the rendering is being done client side while with MVC, the rendering is being server side.
+So which is better? By now you should know that the answer to this question is always "it depends." Both have their advantages and disadvantages, and in this course we'll explore both.
+
+
+## Breaking down MVC further
+
+### Models
+
+First, models are just classes.
+
+Second, these model classes are intended to be a C# representation of a database table.
+
+If Kennel had an actual database, we could imagine an Animal table looking like this
+
+```sql
+CREATE TABLE Animal (
+    Id	integer NOT NULL PRIMARY KEY IDENTITY,
+    [Name]	varchar(80) NOT NULL,
+    Breed	varchar(80),
+    OwnerId	integer NOT NULL,
+    CONSTRAINT FK_AnimalOwner FOREIGN KEY(OwnerId) REFERENCES [Owner](Id)
+);
+```
+
+This table stores the raw data for animals. When you query the database for an animal, you want to represent that data as an object in your code. Below is the most basic representation of that database table as a class.
+
+
+```cs
```

Comment:
I'm not sure that showing code in the chapter is going to make much of an impact at first. Maybe we should live code something right away? The code in this chapter would make more sense after they have some context.

{Edit}
I just saw the second chapter. That's more like what I was thinking.